exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Schema.java

```diff
@@ -16,21 +16,22 @@
 
 package com.exonum.binding.core.service;
 
-import com.exonum.binding.common.hash.HashCode;
-import java.util.List;
+import com.exonum.binding.core.storage.indices.ProofListIndexProxy;
+import com.exonum.binding.core.storage.indices.ProofMapIndexProxy;
 
 /**
  * A schema of the collections (a.k.a. indices) of a service.
+ *
+ * <p>To verify the integrity of the database state on each node in the network,
+ * the core automatically tracks every Merkelized collection used by the user
```

Comment:
AFAIK, not every (e.g. grouped proofs are not tracked).

--- 

Is there much sense in groups of Proof indexes then or shall they be removed completely (from the core also):
they do not affect the state; you cannot create proofs from them. @slowli , WDYT?

Reply:
I left it as it is now. If we decide to implement support for grouped proofs, or to remove them completely, the statement will be true. If we decide to leave it as it is now, the statement must be updated, but I'm not sure how to phrase it (how do we call non-grouped indexes?).

Reply:
> you cannot create proofs from them

You can if these proofs are tied to a more "high-level" proof. This is the case in `cryptocurrency-advanced`, where the history of each wallet is `ProofListIndex`, which is tied to the `state_hash` via the `ProofMapIndex` of wallets. The values of the PMI index (i.e., wallets) contain the `object_hash` of the PLI corresponding to the wallet history. This is clunky and requires effort from service developers, but it works.